### DISCUSSION TITLE
Search within code of repositories

### DISCUSSION THREAD

**Discussion by thorimur** [community/community#60088] [Mobile]
It's currently not possible to search within the code of a repository on mobile (at least iOS). If you want to pull up some random function in a large repo while away from your computer and don't know where it is, you have to open up GitHub in a web browser to search. I find myself wanting to do this a lot.

It would also be helpful to be able to search through everything else, as the search bar on the GitHub website allows, but searching the code of a repo would be a good start.

I noticed that a similar discussion was already created about 2 years ago (#7215). I'm not sure why, but it was mentioned in the replies of that discussion that you're looking for similar feature requests instead of just votes on that feature request. So I thought I'd make another, more recent discussion. :)

**Comment by alcere** [comment]
Hi @thorimur! Code search is one of the top requested features from our users, so I will share your feedback with the team and we will take into account to plan future work. Stay tuned for updates! Thanks!

**Comment by elisealix22** [comment]
Hi @thorimur , we recently published a new version of GitHub Mobile (v1.133.0 and above) that adds code search to repositories. Would love to hear your thoughts! Thank you again for helping us prioritize this feature.

**Reply by thorimur** [reply]
This is great! I really appreciate GitHub prioritizing this feature. :)

If it's welcome, I'd offer thoughts on three potential related features to flesh out the functionality:

1. When viewing the code of a file specifically, it should be possible to find code within that file. Instead of returning a GitHub search result, this should work like <kbd>ctrl</kbd><kbd>F</kbd> does in many applications: highlight each occurrence, provide up and down arrows to scroll between each occurrence, and show a counter "i of totalNumber". (This should work for viewing diffs too.) I think this exists as #4134 .
2. If we navigate to a folder within a repo, the search bar at the top should by default search within that folder, with an option to revert back to global repo search. I'm not sure what the most ergonomic way to communicate this is—maybe the search bar becomes two lines long, and the first line is already filled with a(n abbreviated) `path:path/to/current/folder/` query? But in a way that allows a single backspace to delete this query and return us to a global repo search. I'm sure those with real UX experience at GitHub could figure out the best way to accomplish this. :)
3. I really like the preset query buttons that show up above the keyboard when you start a search (`path`, `symbol`, etc.)! The next step for those could be to offer autocomplete for those queries—e.g. when `path` is tapped (or `path:` is typed), file and folder names from the repo show up to be tapped (and if a folder is tapped, subfolders and files in that folder appear). Obviously this doesn't apply to all queries, but autocomplete for the queries that can be autocompleted would be really neat.

Let me know if any of these would be suitable for a new topic instead of a reply here! (If so, this question should be marked as answered, right?) And again, great work; thanks so much for improving the app! This will make things a lot more convenient. :)

**Comment by Seidahm** [comment]
On
